src/app/contact-list/edit-contact/edit-contact.component.ts

```diff
@@ -0,0 +1,54 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Contact } from 'src/app/contacts.service';
+import { ContactsService } from 'src/app/contacts.service';
+
+@Component({
+  selector: 'app-edit-contact',
+  templateUrl: './edit-contact.component.html',
+  styleUrls: ['./edit-contact.component.css'],
+})
+export class EditContactComponent implements OnInit {
+  contactsService: ContactsService;
+  id: number = -1;
+  user: Contact | undefined;
+  contactForm: FormGroup;
+
+  constructor(
+    private route: ActivatedRoute,
+    private router: Router,
+    contactsService: ContactsService
+  ) {
+    this.contactsService = contactsService;
+    this.contactForm = new FormBuilder().group({
```

Comment:
You're re-instantiating FormBuilder directly within the component, which is not the typical Angular practice. Instead, inject it through the constructor to leverage Angular's dependency injection system